source/common/http/http3/conn_pool.cc

```diff
@@ -173,30 +171,16 @@ allocateConnPool(Event::Dispatcher& dispatcher, Random::RandomGenerator& random_
               "Failed to create Http/3 client. Failed to create quic network connection.");
           return nullptr;
         }
-        // Store a handle to connection as it will be moved during client construction.
-        Network::Connection& connection = *data.connection_;
         auto client = std::make_unique<ActiveClient>(*pool, data);
-        if (connection.state() == Network::Connection::State::Closed) {
```

Comment:
@danzh2010 in future it'd be nice to have an ASSERT(runtime_guard_false) so it's clear the path is safe to delete.